### ISSUE TITLE
Using azure devops sp to authenicate

### ISSUE THREAD

**Issue by gertjvr** [arkiaconsulting/terraform-provider-akc#16]
Feature request, I am using azure devops and the terraform tasks which used the current sp to authenticate, I don't know much about go, but would it be possible to extend the authentication process to use https://github.com/hashicorp/go-azure-helpers and cover all the login scenarios? found the above library as part of https://github.com/terraform-providers/terraform-provider-azurerm/issues/2602

**Comment by arkiaman** [comment]
@gertjvr Hi, as of now the provider handles the CLI and environment authentication process, allowing you to authenticate locally and from a pipeline as well. Which authentication process are you missing ?

**Comment by gertjvr** [comment]
with azure pipeline, I have a service principle setup to execute terraform, but with your provider I have to setup an extra azure cli task and capture the credentials into variable to then use in later steps. If your provider was to use the same env variables that were present for the service principle we could avoid this extra step.

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/service_principal_client_secret

ARM_CLIENT_ID
ARM_CLIENT_SECRET
ARM_SUBSCRIPTION_ID
ARM_TENANT_ID

vs

AZURE_CLIENT_ID
AZURE_CLIENT_SECRET
AZURE_TENANT_ID
AZURE_SUBSCRIPTION_ID

```
      - task: AzureCLI@2
        displayName: setup az cli
        inputs:
          azureSubscription: $(subscription)
          scriptType: 'ps'
          scriptLocation: 'inlineScript'
          inlineScript: |
            Write-Host "##vso[task.setvariable variable=AZURE_CLIENT_ID]$($env:servicePrincipalId)"
            Write-Host "##vso[task.setvariable variable=AZURE_CLIENT_SECRET]$($env:servicePrincipalKey)"
            Write-Host "##vso[task.setvariable variable=AZURE_TENANT_ID]$($env:tenantId)"
            $SUBSCRIPTION_ID = az account show --query id --output tsv
            Write-Host "##vso[task.setvariable variable=AZURE_SUBSCRIPTION_ID]$SUBSCRIPTION_ID"
          addSpnToEnvironment: true
          useGlobalConfig: true
```

**Comment by arkiaman** [comment]
Resolved by #20

**Comment by gertjvr** [comment]
Thank you
